vm/src/builtins/memory.rs

```diff
@@ -865,7 +876,7 @@ struct CastArgs {
     #[pyarg(any)]
     format: PyStrRef,
     #[pyarg(any, optional)]
-    shape: OptionalArg<PyListRef>,
+    shape: OptionalArg<PyObjectRef>,
```

Comment:
If they take only `list` and `tuple`, `Either` works.
```suggestion
    shape: OptionalArg<Either<PyTupleRef, PyListRef>>,
```

If it takes any iterable, `ArgIterable` will fit.

Reply:
Wow good. Let me try `Either`